### PR TITLE
Content Filtering: Don't block requests coming from the site you are on

### PR DESCRIPTION
**Pull request by Kokobe**
Fixes: https://github.com/neevaco/neeva-android/issues/1064

We were blocking requests with the origin (For example `foo.com`) even when we were on that same site (`foo.com`). This PR allows requests with the same origin as the site you are on. 

See code comments for some possible limitations.

### CODE REVIEW

**Review by darinwf — APPROVED** [review]

**Review by chung-lai — COMMENTED** [review]
One thing we may want to follow up is to block cookies for these requests. We do that on iOS to at least provide the basic protection for such requests.

File: components/neeva/renderer/content_filtering_agent.cc [hunk 1]
```diff
@@ -133,6 +133,15 @@ ContentFilteringPolicy ContentFilteringAgent::GetPolicyForRequest(
     return ContentFilteringPolicy::kAllow;
   }
 
+  // There's no point in blocking foo.com from making requests from first-party origins (e.g. foo.com). 
+  // Doing so can break logins or other site functionality. 
+  if (!IsThirdParty(url, first_party_origin)) {
+    if (rules_->mode == mojom::ContentFilterMode::BLOCK_COOKIES) {
```
Comment (Kokobe) [inline components/neeva/renderer/content_filtering_agent.cc]: For First-Party cookies, we want to block them too right? Am I doing this right? @darinwf @chung-lai

Comment (darinwf) [reply components/neeva/renderer/content_filtering_agent.cc]: I think the idea was to map `BLOCK_REQUEST` OR `BLOCK_COOKIES` to `kBlockCookies` instead.

**Review by darinwf — CHANGES_REQUESTED** [review]

File: components/neeva/renderer/content_filtering_agent.cc [hunk 2]
```diff
@@ -133,6 +133,13 @@ ContentFilteringPolicy ContentFilteringAgent::GetPolicyForRequest(
     return ContentFilteringPolicy::kAllow;
   }
 
+  // There's no point in blocking foo.com from making requests from first-party origins (e.g. foo.com). 
+  // Doing so can break logins or other site functionality. 
+  if (!IsThirdParty(url, first_party_origin)) {
+    // Block cookies for first-party cookies to stop them from tracking users too.
```
Comment (darinwf) [inline components/neeva/renderer/content_filtering_agent.cc]: nit: "Block cookies for first-party cookies" doesn't read quite right.

I think the point to make here is that we are downgrading `mode` to `BLOCK_COOKIES` in this case to minimize impact to sites.

**Review by darinwf — APPROVED** [review]

File: components/neeva/renderer/content_filtering_agent.cc [hunk 3]
```diff
@@ -133,6 +133,13 @@ ContentFilteringPolicy ContentFilteringAgent::GetPolicyForRequest(
     return ContentFilteringPolicy::kAllow;
   }
 
+  // There's no point in blocking foo.com from making requests from first-party origins (e.g. foo.com). 
+  // Doing so can break logins or other site functionality. 
+  if (!IsThirdParty(url, first_party_origin)) {
+    // Downgrade from kBlockRequests to kBlockCookies to minimize impact on sites. 
+    return ContentFilteringPolicy::kBlockCookies;
```
Comment (darinwf) [inline components/neeva/renderer/content_filtering_agent.cc:140]: Wait, this is going to break stuff. We only want to apply this to URLs that are blocked by the filter. This is not running the filter at all.

**Pull request merged**